e2e/test/iothub/service/IoTHubCertificateValidationE2ETest.cs

```diff
@@ -18,42 +21,62 @@ public class IotHubCertificateValidationE2ETest : E2EMsTestBase
         [Timeout(TestTimeoutMilliseconds)]
         public async Task ServiceClient_QueryDevicesInvalidServiceCertificateHttp_Fails()
         {
+            // arrange
             using var sc = new IotHubServiceClient(TestConfiguration.IotHub.ConnectionStringInvalidServiceCertificate);
-            var exception = await Assert.ThrowsExceptionAsync<IotHubServiceException>(
-                () => sc.Query.CreateAsync<ClientTwin>("select * from devices")).ConfigureAwait(false);
-
-#if NET472
-            Assert.IsInstanceOfType(exception.InnerException.InnerException.InnerException, typeof(AuthenticationException));
-#else
-            Assert.IsInstanceOfType(exception.InnerException.InnerException, typeof(AuthenticationException));
-#endif
+
+            // act
+            Func<Task> act = async () => await sc.Query.CreateAsync<ClientTwin>("select * from devices").ConfigureAwait(false);
+
+            // assert
+            var error = await act.Should().ThrowAsync<IotHubServiceException>();
+            error.And.StatusCode.Should().Be(System.Net.HttpStatusCode.RequestTimeout);
+            error.And.ErrorCode.Should().Be(IotHubServiceErrorCode.Unknown);
+            error.And.IsTransient.Should().BeTrue();
```

Comment:
Should this be 'BeFals()'?
This is not a transient issue when the cert is invalid, right?

Reply:
This is sortof tricky as our invalid cert testing would throw out `HttpRequestException`, and the cause of such an exception could be transient or intransient, in different cases. Also we have similar problems when the inner exception is a `SocketException`, `WebsocketException`, or `MqttCommunicationTimedOutException`. 

I will bring this question to our standup for more insights and discussions.